drivers/FileBase.cpp

```diff
@@ -52,8 +53,8 @@ FileBase::~FileBase() {
     _mutex->unlock();
 
     if (getPathType() == FilePathType) {
-        extern void remove_filehandle(FileLike *file);
-        remove_filehandle(static_cast<FileLike*>(this));
+        extern void remove_filehandle(FileHandle *file);
```

Comment:
fdopen is a little special, as its the only thing that works on both FILE and FileHandle, so must be implemented in retarget.cpp (or at least it needs knowledge of retarget.cpp internals). It goes in <stdio.h> in POSIX, 

Then there are things like poll() which are non-member things working on 1 or more FileHandles - they go in <unistd.h> in POSIX.

retarget.h feels a bit wrong to me. It feels like it should be the fairly private C library->platform API, rather than something applications should be using. But maybe it's not that?

Wonder where other non-members like poll should go. They're not bound in any way to retarget.cpp, so I had them in FileHandle.cpp and FileHandle.h.